PCbuild/get_external.py

```diff
@@ -9,17 +9,36 @@
 from urllib.request import urlretrieve
 
 
+def retrieve_with_retries(download_location, output_path, reporthook,
+                          max_retries=7):
+    """Download a file with exponential backoff retry and save to disk."""
+    for attempt in range(max_retries):
+        try:
+            resp = urlretrieve(
+                download_location,
+                output_path,
+                reporthook=reporthook,
+            )
+        except ConnectionError as ex:
+            if attempt == max_retries:
```

Comment:
Because `attempt` comes from `range(max_retries)`, this condition can never be hit.

Reply:
🤦‍♀️ thanks for catching that, opened https://github.com/python/cpython/pull/134867